source/client/Controller.py

```diff
@@ -15,42 +15,49 @@ class Controller(ConnectionListener):
 
     def __init__(self, clientState):
         self._state = clientState
-        self.prepared_cards = {} #This is the dict of cards prepared to be played
+        self.prepared_cards = {}     #This is the dict of cards prepared to be played
         self.setName()
         self.ready = False
         self.note = "Game is beginning."
+        # variables needed for games with Shared_Board == True (i.e. Liverpool):
+        self.Meld_Threshold = self._state.rules.Meld_Threshold
+        self.player_index = 0
+        self.visible_scards = [{}] # if Shared_Board both HandView and TableView use data:visible cards.
```

Comment:
Since visible_scards is kind of like played_cards I would think it should belong to clientState and always simply be funneled there so it need not be passed when playing? It isn't ever used here except to funnel it to _state for played cards. Is there a reason for this choice specifically?

Reply:
Turns out that self.visible_scards wasn't set here in the controller ever, it belongs to the two views and hand_view provides it as part of the necessary info to play a card. Simply removing the unnecessary member variable is fine.